File: backend-java/SpringCloudSustainableProducts/shopping-cart-service/src/main/java/be/pxl/services/domain/Product.java
package be.pxl.services.domain;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Product {
    private Long id;
    private String name;
    private String description;
    private BigDecimal price;
//    private Integer quantity;
    private Category category;
    private Score score;
}

File: frontend-web/sustainable-products-app/src/app/models/product.ts
import { Category } from "./category.enum";
import { Score } from "./score.enum";

export class Product {
    id: number | null = null; 
    name: string = ''; 
    description: string = ''; 
    price: number = 0; 
    category: Category | null = null;
    score: Score | null = null; 
    labels: string = ''; 
    quantity: number = 1; 
  }